src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -388,6 +388,7 @@ class %s extends {$options['class']}
     'container.build_hash' => '$hash',
     'container.build_id' => '$id',
     'container.build_time' => $time,
+    'container.runtime_mode' => \PHP_SAPI === 'cli' || \PHP_SAPI === 'phpdbg' ? 'cli' : 'web',
```

Comment:
Could we only check if the SAPI name starts with `cli` but allow suffixes? This will help for alternative CLI implementations. Shouldn't also check for the `embed` SAPI (it behaves more or less like CLI).

Reply:
Updated to add "embed" almost everywhere.
Bad idea for the "cli" prefix: "cli-server" starts with "cli" but serves HTML (php -S)